frontend/src/utils/useBiolink.js

```diff
@@ -43,24 +53,38 @@ export default function useBiolink() {
       obj[strings.nodeFromBiolink(item)] = hierarchy;
       return obj;
     }, {});
-    setHierarchies(newHierarchies);
+    return newHierarchies;
   }
 
   /**
    * Filter out concepts that are not derived classes of base class
    * @returns {array} list of valid concepts
    */
-  function getValidConcepts() {
-    const newConcepts = Object.keys(hierarchies).filter((key) => hierarchies[key].includes(baseClass));
-    setConcepts(newConcepts);
+  function getValidConcepts(allHierarchies) {
+    const newConcepts = Object.keys(allHierarchies).filter((key) => allHierarchies[key].includes(baseClass));
+    return newConcepts;
   }
 
-  function getEdgePredicates() {
-    return Object.entries(biolink.slots).map(([identifier, predicate]) => ({
-      predicate: strings.edgeFromBiolink(identifier),
-      domain: strings.nodeFromBiolink(predicate.domain),
-      range: strings.nodeFromBiolink(predicate.range),
-    }));
+  function dig(obj, keys, target) {
+    let types = [];
+    keys.forEach((key) => {
+      if (obj[key].is_a === target) {
+        types.push(key);
+        types = types.concat(dig(obj, keys, key));
+      }
+    });
+    return types;
+  }
+
+  function getDepthHierarchy(className) {
+    const standardizedClassDictionary = _.transform(biolink.classes,
```

Comment:
Do we have to do this every this function is called?

Reply:
I can move this logic out because we make this object twice. But this function will also only be called once on app load.